stubmodify/modify_test.go

```diff
@@ -78,13 +78,44 @@ func TestWriteAttributionCodeFull(t *testing.T) {
 			t.Error("fileBytes was modified in WriteAttributionCode")
 		}
 
+		// Ensure the size didn't change
```

Comment:
Also, please add a comment on L62 to mention that this `test-stub.exe` file has been modified to include (random?) data in the attribution space in order to verify that we clear everything before writing a new attribution code.

Ideally we should have done that in the test (near L67-68) instead of updating a binary file in the repo but this is done (though I guess extracting the logic to find the moz custom tag in a private function wouldn't be a bad idea and could be reused for that as well as for the checks below).

If you feel like this is something you want to do because it is cleaner, please feel free to re-request a review.

Reply:
I'm happy to do switch it up if you'd rather we stick with the original binary, but I'm curious why you think it's better to modify the binary after reading it rather than make it more like what the binaries will look like after https://bugzilla.mozilla.org/show_bug.cgi?id=1814727 lands?

Reply:
This `test-stub.exe` file is currently the one we can download at https://download-installer.cdn.mozilla.net/pub/firefox/nightly/experimental/bug1318456/test-stub.exe so I thought it'd be good to keep the original version. Now with Bug 1814727, I expect `test-stub.exe` to be updated as well so that's probably fine.

Could you make sure that the test stub file linked above will be updated as part of this bug? Thanks.

Reply:
I don't see any reason to go to the trouble of publishing something like this to archive....the only purpose it would serve is to then copied over here. We could pull in a newer Nightly stub installer after my attribution work lands -- but that won't have a fully non-zero attribution area, so it won't be as good of a test, I think?

Reply:
The `test-stub.exe` file linked above is what gets used/served by the stub attribution service in the nonprod envs so if the prod stub installer changes, it'd be good to adjust the test stub as well so that QA can verify similar behaviors between stage and prod.

Since you changed the `test-stub.exe` here, I suppose it's a matter of getting it re-published in the CDN linked above? This does not have to be fully automated.

Reply:
Oh, TIL. Yes - I'm sure we could do a one-off upload of it to keep them in sync. (I'll note that in prod, the installer changes with every release though...so we're already in a state where production tests don't test the same binary as staging ones.)